### ISSUE TITLE
Performance Benchmark

### ISSUE THREAD

**Issue by michaelsogos** [sergeyzwezdin/Hangfire.Mongo#82]
Dear all,

After reading official documentation, @odinserj describe how much is faster Redis compared to Sql Server.
Do you have some benchmark test to know how much is slower or faster MongoDB compared to Sql Server?

I ask this because in our experience with MongoDB there are some tips and tricks to gain performance querying MongoDB (from c# of course), without knowing it MongoDB can be slower than Sql Server in specific situation.

Thanks in advance for your time :)

**Comment by gottscj** [comment]
@michaelsogos,
 We dont have any benchmark tests. Sorry.
I would think you would have similar experience.
We are working on a ReactiveMongoStorage which will give better performance as we dont need to poll the job queue

**Comment by MarLoe** [comment]
@michaelsogos,
As @gottscj  says, we don't have any benchmarks. :(
If you could help us create some, that would be awesome.
That being said, we are preparing release 0.5.5 that has PR #109 that should optimize performance.

In release 0.6.0 (which has been on it way for quite a while now), we will change the polling strategy with a trailing cursor strategy that allows us for much quicker response times when it comes to the time from a job is enqueued to it is executed.

**Comment by michaelsogos** [comment]
Dear @MarLoe,

We moved on redis because with mongo we had a lot of issue with simple examples, like we cannot retry a job because some "document duplication" issue. For these and other reasons, of course after many test in many environment, we found redis fast, free of bug, easy to maintain and easy to scale.

Actually we stopped using mongodb and we didn't made load and stress test, so i'm very sorry but i cannot supply valueable information.
